addons/mail/models/mail_guest.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+import uuid
+from odoo.tools import consteq
+from odoo import fields, models, _
+from odoo.addons.base.models.res_partner import _tz_get
+
+class MailGuest(models.Model):
+    _name = 'mail.guest'
+    _description = "Guest"
+
+    access_token = fields.Char(
+        copy=False,
+        default=lambda self: str(uuid.uuid4()),
+        groups='base.group_system',
+        readonly=True,
+        required=True,
+    )
+    channel_ids = fields.Many2many(comodel_name='mail.channel')
+    country_id = fields.Many2one(string="Country", comodel_name='res.country')
+    lang_id = fields.Many2one(string="Language", comodel_name='res.lang')
+    name = fields.Char(required=True)
+    timezone = fields.Selection(selection=_tz_get)
+
+    def format_guests(self):
```

Comment:
why not use `name_get`?

Reply:
It will likely contain more stuff in the result I think, need to double check after PR clean.